frame.js

```diff
@@ -0,0 +1,76 @@
+class Frame {
+
+  constructor() {
+    this.rolls = [];
+    this.scoringRolls = [];
+    this.maxRolls = 2;
+    this.bonus = 0;
+    this.extra = false;
+  }
+
+  roll(pins) {
+    this.rolls.push(pins);
+    this.scoringRolls.push(pins);
+  }
+
+  nonScoringRoll(pins) {
+    this.rolls.push(pins);
+  }
+
+  getRolls() {
+    return [...this.rolls];
```

Comment:
amazing, i didnt know ...array existed :)

Reply:
Haha, I didn't know it either but I found it out by looking up how to duplicate an array in Javascript (like array.dup in Ruby)
Found it here: https://www.freecodecamp.org/news/how-to-clone-an-array-in-javascript-1d3183468f6a/